src/dbnode/storage/index/results.go

```diff
@@ -156,10 +157,25 @@ func (r *results) Reset(nsID ident.ID) {
 }
 
 func (r *results) Finalize() {
+	if r.noFinalize {
+		return
+	}
+
 	r.Reset(nil)
 
 	if r.pool == nil {
 		return
 	}
 	r.pool.Put(r)
 }
+
+func (r *results) NoFinalize() {
```

Comment:
Hm, I can't actually see where this is used - is this still used?

Reply:
No, I just had already written and tested it and figure'd we'd want it for query caching later so might as well leave it in. Want me to take it out?